### PR TITLE
[WIP] Api compatibility with legacy content objects

### PR DESCRIPTION
**Pull request by rossriley**
As agreed this PR adds one method to each of the Repeater collection and Repeater field objects.

It matches the API of dealing with existing content items.

### CODE REVIEW

**Comment by bobdenotter** [comment]
`getdecodedValue()` works the other way around from #6472 now. :-)

screen shot 2017-03-11 at 12 25 37

**Comment by bobdenotter** [comment]
I can't get `fieldtype` to work yet.. 

```
        {% for repeater in record.repeater %}
            {% for key, repeaterfield in repeater %}
                {{ repeater.fieldtype(key) }}
            {% endfor %}
        {% endfor %}
```

gives: 

screen shot 2017-03-11 at 12 26 30

**Comment by rossriley** [comment]
Sorry, forgot we have two classes now. Added to the LazyCollection too.

**Comment by bobdenotter** [comment]
Now i'm getting this: 

screen shot 2017-03-11 at 12 41 37

Would it help if i threw pasted the contenttype and template to reproduce?

**Comment by bobdenotter** [comment]
I'm using the default `showcase` contenttype, and the template looks like this: 

```
{% extends 'partials/_master.twig' %}

{% block main %}

    <h1>{{ record.title }}</h1>

    {% for repeater in record.repeater %}

        {% for key, repeaterfield in repeater %}

            {% set fieldtype = repeater.fieldtype(key) %}
            {% set value = repeater.getDecodedValue(key) %}

            <p>field is: {{ fieldtype }}</p>
            {{ dump(value) }}

        {% endfor %}

    {% endfor %}
```

**Comment by rossriley** [comment]
OK, try again.

**Comment by bobdenotter** [comment]
Halfway there. :-)

`{% set fieldtype = repeater.fieldtype(key) %}` -> works
`{% set value = repeater.decodedValue(key) %}` -> gives error: 

screen shot 2017-03-11 at 13 01 44

**Comment by bobdenotter** [comment]
Almost perfect. :-)

`{{ repeater.decodedValue(key) }}` still does the same thing as `{{ repeater.get(key) }}`, as in it returns escaped HTML: 

screen shot 2017-03-11 at 12 25 37

**Comment by GawainLynch** [comment]
Also if you're pushing more, gentle nudge on the PHPDoc types :wink:

**Comment by bobdenotter** [comment]
I wouldn't mind adding the PHPDoc myself in a consecutive PR. 

Doe ik ook nog eens wat. ;-)

**Comment by GawainLynch** [comment]
Your call … just thinking about it, these should be properly marked up as `@deprecated`, no?

**Comment by bobdenotter** [comment]
Yes, they should be marked `@deprecated` too. :-)

**Comment by CarsonF** [comment]
Also missed the interface.

**Comment by GawainLynch** [comment]
Also wondering if they should be marked `@internal` as otherwise we're something locking into the external API that a) never worked; b) is scheduled to be ripped out inside the 3.x lifecycle

File: src/Storage/Field/Collection/LazyFieldCollection.php [hunk 1]
```diff
@@ -68,6 +68,33 @@ public function setGrouping($grouping)
...
+     *  This can be removed once the deprecation of legacy content is complete.
+     *
+     * @param $fieldName
+     * @return mixed
+     */
+    public function getDecodedValue($fieldName)
+    {
+        $this->initialize();
+
+        return $this->collection->getDecodedValue($fieldName);
```
Comment (GawainLynch) [inline src/Storage/Field/Collection/LazyFieldCollection.php]: Shouldn't this just:

```php
return $this->collection->get($fieldName);
```

**Comment by GawainLynch** [comment]
@bobdenotter pulled this into #6474, so closing

**Pull request closed without merging**